File: backend/app/schemas/consultation.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field

class ConsultationCreateRequest(BaseModel):
    recipient_age_group: str
    recipient_gender: str
    relationship: str
    purpose: str
    budget_min: Optional[int] = None
    budget_max: Optional[int] = None
    hobbies: List[str] = Field(default_factory=list)
    avoid_items: List[str] = Field(default_factory=list)
    desired_mood: str
    note: str = ""


class ConsultationCreateResponseData(BaseModel):
    consultation_id: str
    input: Dict[str, Any]
    result: Dict[str, Any]
    created_at: Optional[datetime] = None


class ConsultationCreateResponse(BaseModel):
    data: ConsultationCreateResponseData
    message: str = "success"


# 一覧の中の1件分のデータ構造
class ConsultationListItem(BaseModel):
    id: str
    title: str
    is_favorite: bool
    visibility: str
    created_at: datetime

    class Config:
        from_attributes = True

# dataフィールドの中身の構造
class ConsultationListResponseData(BaseModel):
    items: List[ConsultationListItem]
    page: int
    limit: int
    total: int

# 最終的な共通レスポンス構造
class ConsultationListResponse(BaseModel):
    data: ConsultationListResponseData
    message: str = "success"

# 詳細表示用のデータ構造
class ConsultationDetailResponseData(BaseModel):
    id: str
    title: str
    input_conditions: Dict[str, Any]
    ai_response: Dict[str, Any]
    is_favorite: bool
    visibility: str
    created_at: datetime

    class Config:
        from_attributes = True

# 最終的な共通レスポンス構造
class ConsultationDetailResponse(BaseModel):
    data: ConsultationDetailResponseData
    message: str = "success"

File: frontend/src/types/consultation.ts
export type Visibility = "private" | "public" | "unlisted";

export type ConsultationInput = {
  recipient_age_group: string;
  recipient_gender: string;
  relationship: string;
  purpose: string;
  budget_min: number;
  budget_max: number;
  hobbies: string[];
  avoid_items: string[];
  desired_mood: string;
  note: string;
};

export type GiftCandidate = {
  name: string;
  reason: string;
  budget_range: string;
  caution: string;
  suitable_for: string;
  message: string;
};

export type ConsultationResult = {
  summary: string;
  gift_candidates: GiftCandidate[];
  tips: string[];
  avoid: string[];
};

export type ConsultationCreateResponse = {
  consultation_id: string;
  input: ConsultationInput;
  result: ConsultationResult;
  created_at: string;
};

export type ConsultationListItem = {
  id: string;
  title: string;
  is_favorite: boolean;
  visibility: Visibility;
  created_at: string;
};

export type ConsultationDetail = {
  id: string;
  title: string;
  input: ConsultationInput;
  result: ConsultationResult;
  is_favorite: boolean;
  visibility: Visibility;
  created_at: string;
};

export type ConsultationUpdateInput = {
  is_favorite?: boolean;
  visibility?: Visibility;
  title?: string;
};

export type ConsultationUpdateResponse = {
  consultation_id: string;
  is_favorite: boolean;
  visibility: Visibility;
  title: string;
  updated_at?: string;
};

export type ConsultationDeleteResponse = {
  consultation_id: string;
};
